ants_test.go

```diff
@@ -427,24 +427,26 @@ func TestMaxBlockingSubmit(t *testing.T) {
 
 func TestNonblockingSubmitWithFunc(t *testing.T) {
 	poolSize := 10
-	ch1 := make(chan struct{})
+	var wg sync.WaitGroup
 	p, err := NewPoolWithFunc(poolSize, func(i interface{}) {
 		longRunningPoolFunc(i)
-		close(ch1)
+		wg.Done()
 	}, WithNonblocking(true))
 	assert.NoError(t, err, "create TimingPool failed: %v", err)
 	defer p.Release()
+	ch := make(chan struct{})
 	for i := 0; i < poolSize-1; i++ {
-		assert.NoError(t, p.Invoke(nil), "nonblocking submit when pool is not full shouldn't return error")
+	        wg.Add(1)
```

Comment:
Move this line up and change it to `wg.Add(poolSize-1)`

Reply:
done